### ISSUE TITLE
Converting GRCh38 cache files

### ISSUE THREAD

**Issue by sigven** [Ensembl/ensembl-vep#289]
Hi,

I just downloaded the latest human cache files (Ensembl release 94):
ftp://ftp.ensembl.org/pub/release-94/variation/VEP/homo_sapiens_vep_94_GRCh37.tar.gz
ftp://ftp.ensembl.org/pub/release-94/variation/VEP/homo_sapiens_vep_94_GRCh38.tar.gz

I converted the GRCh37 cache successfully with `convert_cache.pl`, but the GRCh38 did not recognize that any conversion was necessary, is this correct?

`2018-10-03 06:47:12 - Processing homo_sapiens`
`2018-10-03 06:47:12 - Processing version 94_GRCh38`
`2018-10-03 06:47:13 - No unprocessed types remaining, skipping`
`2018-10-03 06:47:13 - All done!`

regards,
Sigve

**Comment by at7** [comment]
Hi Sigve,

I missed another required file edit. You also need to update the following line:

variation_cols  chr,variation_name,failed,somatic,start,end,allele_string,strand,minor_allele,minor_allele_freq,clin_sig,phenotype_or_disease,pubmed,AFR,AMR,EAS,EUR,SAS,AA,EA,gnomAD,gnomAD_AFR,gnomAD_AMR,gnomAD_ASJ,gnomAD_EAS,gnomAD_FIN,gnomAD_NFE,gnomAD_OTH,gnomAD_SAS


Can you please remove chr, from the start of the column names. It should look like this after the edit:

variation_cols variation_name,failed,somatic,start,end,allele_string,strand,minor_allele,minor_allele_freq,clin_sig,phenotype_or_disease,pubmed,AFR,AMR,EAS,EUR,SAS,AA,EA,gnomAD,gnomAD_AFR,gnomAD_AMR,gnomAD_ASJ,gnomAD_EAS,gnomAD_FIN,gnomAD_NFE,gnomAD_OTH,gnomAD_SAS

And then run the convert_cache script again.

The corrected files will be on the FTP site at the latest tomorrow!

Thank you,
Anja

**Comment by dbolser** [comment]
I'm seeing the same error (110).

When I delete the line that reads `var_type  tabix` I get a different error:

```
Illegal division by zero at /home/danbolser/Build/ensembl-vep/convert_cache.pl line 555.
```

I tried editing the variation_cols as suggested above, but I get the same error.

**Comment by dbolser** [comment]
Yup, 110.

Right, I thought this might be true, but the docs didn't mention this, and when I saw the comment above, "still need to be converted", I thought I'd better ask :-)

Thank you for explaining.

Can you update the documentation here:
https://www.ensembl.org/info/docs/tools/vep/script/vep_cache.html#convert


Sincerely,

**Comment by dglemos** [comment]
Thanks for pointing it out. We will make it clear in the docs there is no need to convert the cache. 

Best wishes,
Diana

**Comment by dbolser** [comment]
Having suffered slow running times, I've come back to double check the cache...

I'm sure there is a better way to do this, but I simply decided to check if `convert_cache.pl` 'does anything' when I point it at an older release. It always seems to 'do nothing':

```
dan@8fddb486abc8:/raid/general/Data/VEP$ /opt/vep/src/ensembl-vep/convert_cache.pl --dir Cache/ --species homo_sapiens --version 109_GRCh38
2023-11-15 12:22:06 - Processing homo_sapiens
2023-11-15 12:22:06 - Processing version 109_GRCh38
2023-11-15 12:22:06 - No unprocessed types remaining, skipping
2023-11-15 12:22:06 - All done!
dan@8fddb486abc8:/raid/general/Data/VEP$ /opt/vep/src/ensembl-vep/convert_cache.pl --dir Cache/ --species homo_sapiens --version 109_GRCh37
2023-11-15 12:23:23 - Processing homo_sapiens
2023-11-15 12:23:23 - Processing version 109_GRCh37
2023-11-15 12:23:23 - No unprocessed types remaining, skipping
2023-11-15 12:23:23 - All done!
```

This doesn't fill me with confidence that it's correctly 'doing nothing' on `110_GRCh38`...

Ah... sorry, I managed to not notice the difference in the link you provided, I should have been using [/pub/release-110/variation/**indexed_vep_cache**/](http://ftp.ensembl.org/pub/release-110/variation/indexed_vep_cache/) instead of [http://ftp.ensembl.org/pub/release-110/variation/**vep**/](http://ftp.ensembl.org/pub/release-110/variation/vep/)...

I thought you were just adding the indexes to the regular location...

The above should help me, but I still /think/ that convert_cache.pl isn't behaving...


Cheers,

**Comment by dglemos** [comment]
We provided the indexed and non-indexed cache files on the FTP.
They are available in different locations:
- indexed: http://ftp.ensembl.org/pub/release-110/variation/indexed_vep_cache/
- non-indexed: http://ftp.ensembl.org/pub/release-110/variation/vep/

The script convert_cache.pl is redundant now - as we provide the caches, we no longer advise using it.
As I mentioned before, we are going to improve the documentation.

Best wishes,
Diana

**Comment by dbolser** [comment]
Thanks Diana,

If it's redundant (and documented) that's fine.


Cheers,

**Comment by dbolser** [comment]
Out of interest, who benefits from the un-indexed files?

**Comment by dglemos** [comment]
Indexed caches require Bio::DB::HTS or tabix installed. Non-indexed allows users to keep using the cache even though they don't have any of those modules installed.

**Comment by dbolser** [comment]
I see, thanks @dglemos.

It's now running *slower* with the indexed cache... 😅

**Comment by dglemos** [comment]
Can you send the command you run with the indexed cache and with non-indexed?
Did you download the cache from here http://ftp.ensembl.org/pub/release-110/variation/indexed_vep_cache/?
In your indexed cache, do you have the following files `all_vars.gz` and `all_vars.gz.csi` inside the chromosome folders?

**Comment by dbolser** [comment]
Yes, I double checked the download with the CHECKSUM and by looking for
different files.

Unfortunately the `...summary.html` hides options that are set within
`--config my_file.config`.

I'm just going to see if I can output those with a couple of small code
changes then I'll get back to you. It's possible I changed something else.
The difference I'm seeing is 700 variants per minute vs. 3500 per minute,
which I believe is way below what I should be expecting anyway...

Some options I haven't changed are:
```
offline             true
cache               true
dir                 Cache
species             homo_sapiens
assembly            GRCh38

fasta               Cache/Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz
check_ref           true
format              ensembl
compress_output     bgzip
force_overwrite     true
```

but I have 'played with` a few other options, so I'll double check that.

I'll also turn off `check_ref` for 'production' (once I finish testing).


Cheers,
